Landing pages Home

chu4 opened the pull request:
Added landing Layout for home Controller
with new header + used existing session new View for the filling of the page
![image](https://user-images.githubusercontent.com/23340152/34469189-fde2a02a-ef21-11e7-8bb7-b58e8c62073d.png)

egorodessa commented on app/assets/stylesheets/landing.scss:
```diff
@@ -0,0 +1,13 @@
+.home-back-image{
+  color: rgba(255,255,255,0.6);
+  background: linear-gradient(rgba(0,0,0,.5), rgba(0,0,0,.5)),url('https://www.kenresearch.com/blog/wp-content/uploads/2017/11/Jordan-telecommunications-market.jpg');
```
download pic, at least for speed and in case it will be not available

igorkasyanchuk replied:
of course download. ALL resources hosted locally in your app.

igorkasyanchuk commented on app/controllers/home_controller.rb:
```diff
@@ -1,3 +1,4 @@
 class HomeController < ApplicationController
+  layout "landing"
```
this is good

igorkasyanchuk replied:
+ create an empty pages for Pricing, etc
Just empty page with `h1 Pricing` for example

igorkasyanchuk commented on app/views/layouts/_header.html.slim:
```diff
@@ -1,7 +1,7 @@
   .navbar-wrapper
     .navbar-container.content
-      = link_to 'SalesAssistant', new_session_path(resource_name), id: 'logo'
+      = link_to 'SalesAssistant', "#", id: 'logo'
```
що за прикол з лінками які нікуди не ведуть?
лого завжди веде на головну сторінку, таке правило

chu4 replied:
Потрібна головна сторінка після логування, якись аккаунт чи юзер пейдж. Був прописаний рут який зараз сторінка лендінгу

igorkasyanchuk replied:
це мало би бути дашборд (чи той ПР ще не замеоджаний?)

igorkasyanchuk commented on app/views/layouts/_header.html.slim:
```diff
@@ -44,11 +44,11 @@ nav.header-navbar.navbar-expand-md.navbar.navbar-with-menu.navbar-static-top.nav
-                  i.fa.fa-user
-                  | Log in
\ No newline at end of file
+      /- else
```
а це кому лишив комент?

igorkasyanchuk commented on app/views/layouts/landing.html.slim:
```diff
@@ -0,0 +1,13 @@
+    = stylesheet_link_tag    'application', media: 'all'
+    = javascript_include_tag 'application'
+  body
+    = render 'layouts/header_landing'
```
перенеси цей паршал в /shared/header

igorkasyanchuk commented on app/views/layouts/landing.html.slim:
```diff
@@ -0,0 +1,13 @@
+    = render 'layouts/header_landing'
+    = yield
+    = any_login_here if defined?(AnyLogin)
+    = render 'layouts/footer'
```
і футер теж

chu4 commented:
Changed to a light header/footer + different local background image
Added root_path for authenticated user
![image](https://user-images.githubusercontent.com/23340152/34483697-efd3b83c-efc8-11e7-8785-944c038d44d7.png)

chu4 commented:
New container look
![image](https://user-images.githubusercontent.com/23340152/34488946-1ded5376-efe3-11e7-86e7-afcab4b4effd.png)

Kmalynss commented on app/controllers/home_controller.rb:
```diff
@@ -1,3 +1,11 @@
+
+  def pricing; end
+
+  def contacts; end
```
Do we still need this empty actions? AFAIK there is no need in them if there are templates with appropriate names.

igorkasyanchuk replied:
very simple templates must be added
with h1 Pricing for example

Kmalynss replied:
I understand. I'm saying that we don't need those definitions if we have templates. As i remember.

chu4 replied:
Yes works without actions in the controller

chu4 commented:
New version
![image](https://user-images.githubusercontent.com/23340152/34497620-f9803b1a-f006-11e7-93ba-f3e845c8aa62.png)
![image](https://user-images.githubusercontent.com/23340152/34497625-fe9ec6d4-f006-11e7-9809-3ead72416395.png)

chu4 commented:
-Resolved width bug
-Added Raleway font
-Changed the main background image as negotiated with the team today
-Refactored text to the suggested example
![image](https://user-images.githubusercontent.com/23340152/34526553-7ac08190-f0ab-11e7-9d6a-081fd9a83f46.png)
![image](https://user-images.githubusercontent.com/23340152/34526571-875e5706-f0ab-11e7-8a72-2367d7a9be1d.png)

chu4 commented:
add landing page accecable for authenticated user with different header(dashbooard + logout)
find better font family
think about header

chu4 commented:
**CHECK LIST**
- [ ] specs added
- [ ] specs passed
- [ ] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [ ] PR is reviewed manually again (to make sure you have 100% ready code)
- [ ] PR doesn't have conflicts with main branch
- [ ] I've checked new feature as logged in and logged out user if needed
- [ ] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added

![screenshot from 2018-01-07 17-11-39](https://user-images.githubusercontent.com/23340152/34651021-14c31ca6-f3d3-11e7-977e-1218e3fb78f0.png)

- [ ] screenshot of feature is added if task has UI

not signed_in
![screenshot from 2018-01-07 17-24-50](https://user-images.githubusercontent.com/23340152/34651024-2da02c6e-f3d3-11e7-8c06-f93aec420f40.png)
![screenshot from 2018-01-07 17-24-59](https://user-images.githubusercontent.com/23340152/34651025-2dc42d44-f3d3-11e7-9818-529d83476d44.png)

signed_in
![screenshot from 2018-01-07 17-26-57](https://user-images.githubusercontent.com/23340152/34651031-3ab79d74-f3d3-11e7-9bc9-12a2bf019f21.png)
![screenshot from 2018-01-07 17-27-02](https://user-images.githubusercontent.com/23340152/34651032-3ad97e30-f3d3-11e7-9252-310d60104735.png)

igorkasyanchuk commented:
For landing layout do not show header with dashboard, companies, etc. Just show everything as for not logged in user. Basically everything is the same except sign up and sign in links are not visible.

chu4 commented:
Should i just add log out instead of sign in&up?

igorkasyanchuk commented:
What you have now is OK. But where is the of the links in header. Pricing, contact, etc.

chu4 commented:
for signed in user i moved those to footer. so that landing links wont mess with pages available for signed in user

igorkasyanchuk commented on app/views/layouts/shared/header/_header_landing.html.slim:
```diff
@@ -0,0 +1,49 @@
+      li.nav-item
+        = link_to "CONTACTS", contacts_path, class:"nav-link font-medium-2"
+  - if user_signed_in?
+    .horizontal-menu.navbar-container
```
don't we have such code already in other layout?
extract to partial.
this dropdown when user is logged in.
Also "arrow" next to username in this layout looks not aligned well.

chu4 replied:
edited the arrow and made a partial for that bit of a header
![screenshot from 2018-01-08 22-21-30](https://user-images.githubusercontent.com/23340152/34692262-3d33b16e-f4c8-11e7-9e27-c3b43b6dcfdc.png)

igorkasyanchuk commented on app/assets/stylesheets/css/bootstrap-extended.css:
```diff
@@ -490,8 +490,8 @@ Bootstrap Enhance */
 
-.navbar-horizontal .dropdown-toggle::after {
-  top: -9px !important; }
+/*.navbar-horizontal .dropdown-toggle::after {*/
```
how this change affected admin layout? did you test it?

chu4 replied:
this style made arrow not be alined

chu4 replied:
application layout header is not affected with that change
![screenshot from 2018-01-09 13-13-59](https://user-images.githubusercontent.com/23340152/34718447-5648d7de-f53f-11e7-99a6-adb4c4432c09.png)

igorkasyanchuk replied:
why you keep this comment in CSS? do you need it? IF NOT - remove

chu4 commented:
last comment is a possible fix for production. Bug where header didn't render properly

igorkasyanchuk commented on app/assets/stylesheets/landing.scss:
```diff
@@ -0,0 +1,56 @@
+.land-font{
```
add space before "{"

igorkasyanchuk commented on app/assets/stylesheets/landing.scss:
```diff
@@ -0,0 +1,56 @@
+  text-align: center;
+}
+
+.landing-cont-back3{
```
and here ... and in other places

igorkasyanchuk commented on app/views/layouts/shared/header/_header_landing.html.slim:
```diff
@@ -0,0 +1,27 @@
+  .horizontal-menu.navbar-container.ml-0
+    ul.nav.navbar-nav data-menu="menu-navigation"
+      li.nav-item
+        = link_to "HOME", home_path, class:"nav-link font-medium-2"
```
for upper case use CSS for navigation in main layout

chu4 commented:
![image](https://user-images.githubusercontent.com/23340152/34773027-e5925e28-f612-11e7-81b2-645a371ba416.png)
![image](https://user-images.githubusercontent.com/23340152/34773045-ef7e4910-f612-11e7-8475-ef73aea19686.png)
